### ISSUE TITLE
Potential numerical instability in "categorical_crossentropy" when not computing from logits?

### ISSUE THREAD

**Issue by zerolocker** [keras-team/keras#6338]
I was wondering if  a "from_logits" parameter could be added to the loss *categorical_crossentropy. In tensorflow, there is an implementation I am talking about: tf.nn.softmax_cross_entropy_with_logits

Right now, most model definition examples in  Keras document (e.g. [here](https://keras.io/getting-started/functional-api-guide/)) uses a softmax  activation on top of the final output of the layer. But we know the gradient computation will be more numerically instable, as suggested by [tf.nn.softmax_cross_entropy_with_logits](https://www.tensorflow.org/versions/master/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits)

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
